main/boofcv-ip/src/test/java/boofcv/alg/misc/TestImageBandMath.java

```diff
@@ -29,53 +29,354 @@
 import java.util.Random;
 
 /**
- * @author Peter Abeles
+  * @author Nico
  */
-public class TestImageBandMath extends CompareIdenticalFunctions {
+public class TestImageBandMath {
```

Comment:
Can you revert this file back to the version found in 945d70f343579c4aea6ce8ca5b20d7889b0bbb1d ?

That code does a comparison against ImplImageBandMath, which is what ImageBandMath is a wrapper around. ImplImageBandMath has behavior unit tests.

One addition that would be nice would be to verify that the regular and MT version is called correctly depending on the value of USE_CONCURRENT. I've not done that myself because doing it is a pain and requires overloading a static function using reflection magic. I think a tool like mockito let's you do that but I've never used mockito.

Reply:
I tried, but when running that code I get a number of  errors, probably caused the addition of function overloading.  I wrote a version that makes sense to me, but run into errors that I do not understand, and there is too much happening behind the scenes for me to figure out how to make it work.   I ended up with the code below, but get a runtime exception
```suggestion
@Override
	protected Object[][] createInputParam(Method candidate, Method validation) {

		Class[] types = candidate.getParameterTypes();
		int outputIdx = 1;

		Object[][] output = new Object[2][types.length];

		for (int i = 0; i < output.length; i++) {
			Object[] param = output[i];
			int paramNr = 0;

			param[paramNr++] = new Planar(types[outputIdx],width,height,numBands);
			param[paramNr++] = GeneralizedImageOps.createSingleBand(types[outputIdx],width,height);
			if (candidate.getName().equals("stdDev")) {
				param[paramNr++] = null;
			}
			if (candidate.getParameterCount() > 3) {
				param[paramNr++] = 1;
				param[paramNr++] = numBands - 2 + i; // test even and off bands. this is inclusive
			}
			GImageMiscOps.fillUniform((Planar)param[0],rand,0,100);
			GImageMiscOps.fillUniform((ImageGray)param[1],rand,0,100);
		}

		return output;
	}
```

Reply:
Now more or less reverted to the original version.  The problems were caused by calling the implementation of the stdDev functions with avg set to null.  There is now a bit of redundant code (the implementation should never have to generate an avg image as this code should only be called from boofcv itself, but oh well).